### ISSUE TITLE
Getting silent NotSerializableException when using getFromResource

### ISSUE THREAD

**Issue by Svantulden** [wonson/androidsvg#1]
Hi, 

First of all, thanks for forking the androidsvg library with support for modifying the DOM tree!! It is exactly what I need for my Android project (changing fill colors for specific elements). 

Now, when trying out your fork, I use this code: 
```Java
holder.locationIconImageView.setLayerType(View.LAYER_TYPE_SOFTWARE, null);
try
{
      SVG svg = SVG.getFromResource(getActivity(), R.raw.pinpoint);
      Drawable drawable = new PictureDrawable(svg.renderToPicture());
      holder.locationIconImageView.setImageDrawable(drawable);
}
catch(SVGParseException e)
{
      Log.d("SVGPARSE", "" + e.getMessage());
}
```

But for every time this code gets called, I am getting a silent `java.io.NotSerializableException: android.graphics.Matrix` throw. I have traced the problem back to [line 2096 in SVG.java](https://github.com/wonson/androidsvg/blob/master/libandroidsvg/src/main/java/com/caverock/androidsvg/SVG.java#L2096), but I don't know what exactly is wrong there to throw this. 

Could you take a look at this? 

Here is the stacktrace: 

```
03-12 09:12:15.792    8648-8648/example.app W/System.err﹕ java.io.NotSerializableException: android.graphics.Matrix
03-12 09:12:15.793    8648-8648/example.app W/System.err﹕ at java.io.ObjectOutputStream.writeNewObject(ObjectOutputStream.java:1344)
03-12 09:12:15.793    8648-8648/example.app W/System.err﹕ at java.io.ObjectOutputStream.writeObjectInternal(ObjectOutputStream.java:1651)
03-12 09:12:15.793    8648-8648/example.app W/System.err﹕ at java.io.ObjectOutputStream.writeObject(ObjectOutputStream.java:1497)
03-12 09:12:15.793    8648-8648/example.app W/System.err﹕ at java.io.ObjectOutputStream.writeObject(ObjectOutputStream.java:1461)
03-12 09:12:15.793    8648-8648/example.app W/System.err﹕ at java.io.ObjectOutputStream.writeFieldValues(ObjectOutputStream.java:959)
03-12 09:12:15.793    8648-8648/example.app W/System.err﹕ at java.io.ObjectOutputStream.defaultWriteObject(ObjectOutputStream.java:360)
03-12 09:12:15.793    8648-8648/example.app W/System.err﹕ at com.caverock.androidsvg.SVG$GraphicsElement.writeObject(SVG.java:2096)
03-12 09:12:15.793    8648-8648/example.app W/System.err﹕ at java.lang.reflect.Method.invoke(Native Method)
03-12 09:12:15.793    8648-8648/example.app W/System.err﹕ at java.lang.reflect.Method.invoke(Method.java:372)
03-12 09:12:15.793    8648-8648/example.app W/System.err﹕ at java.io.ObjectOutputStream.writeHierarchy(ObjectOutputStream.java:1033)
03-12 09:12:15.793    8648-8648/example.app W/System.err﹕ at java.io.ObjectOutputStream.writeNewObject(ObjectOutputStream.java:1384)
03-12 09:12:15.793    8648-8648/example.app W/System.err﹕ at java.io.ObjectOutputStream.writeObjectInternal(ObjectOutputStream.java:1651)
03-12 09:12:15.793    8648-8648/example.app W/System.err﹕ at java.io.ObjectOutputStream.writeObject(ObjectOutputStream.java:1497)
03-12 09:12:15.793    8648-8648/example.app W/System.err﹕ at java.io.ObjectOutputStream.writeObject(ObjectOutputStream.java:1461)
03-12 09:12:15.793    8648-8648/example.app W/System.err﹕ at java.util.ArrayList.writeObject(ArrayList.java:648)
03-12 09:12:15.793    8648-8648/example.app W/System.err﹕ at java.lang.reflect.Method.invoke(Native Method)
03-12 09:12:15.793    8648-8648/example.app W/System.err﹕ at java.lang.reflect.Method.invoke(Method.java:372)
03-12 09:12:15.793    8648-8648/example.app W/System.err﹕ at java.io.ObjectOutputStream.writeHierarchy(ObjectOutputStream.java:1033)
03-12 09:12:15.793    8648-8648/example.app W/System.err﹕ at java.io.ObjectOutputStream.writeNewObject(ObjectOutputStream.java:1384)
03-12 09:12:15.793    8648-8648/example.app W/System.err﹕ at java.io.ObjectOutputStream.writeObjectInternal(ObjectOutputStream.java:1651)
03-12 09:12:15.793    8648-8648/example.app W/System.err﹕ at java.io.ObjectOutputStream.writeObject(ObjectOutputStream.java:1497)
03-12 09:12:15.793    8648-8648/example.app W/System.err﹕ at java.io.ObjectOutputStream.writeObject(ObjectOutputStream.java:1461)
03-12 09:12:15.793    8648-8648/example.app W/System.err﹕ at java.io.ObjectOutputStream.writeFieldValues(ObjectOutputStream.java:959)
03-12 09:12:15.793    8648-8648/example.app W/System.err﹕ at java.io.ObjectOutputStream.defaultWriteObject(ObjectOutputStream.java:360)
03-12 09:12:15.793    8648-8648/example.app W/System.err﹕ at com.caverock.androidsvg.SVG$SvgConditionalContainer.writeObject(SVG.java:1864)
03-12 09:12:15.793    8648-8648/example.app W/System.err﹕ at java.lang.reflect.Method.invoke(Native Method)
03-12 09:12:15.793    8648-8648/example.app W/System.err﹕ at java.lang.reflect.Method.invoke(Method.java:372)
03-12 09:12:15.793    8648-8648/example.app W/System.err﹕ at java.io.ObjectOutputStream.writeHierarchy(ObjectOutputStream.java:1033)
03-12 09:12:15.793    8648-8648/example.app W/System.err﹕ at java.io.ObjectOutputStream.writeNewObject(ObjectOutputStream.java:1384)
03-12 09:12:15.793    8648-8648/example.app W/System.err﹕ at java.io.ObjectOutputStream.writeObjectInternal(ObjectOutputStream.java:1651)
03-12 09:12:15.793    8648-8648/example.app W/System.err﹕ at java.io.ObjectOutputStream.writeObject(ObjectOutputStream.java:1497)
03-12 09:12:15.793    8648-8648/example.app W/System.err﹕ at java.io.ObjectOutputStream.writeObject(ObjectOutputStream.java:1461)
03-12 09:12:15.793    8648-8648/example.app W/System.err﹕ at java.util.ArrayList.writeObject(ArrayList.java:648)
03-12 09:12:15.793    8648-8648/example.app W/System.err﹕ at java.lang.reflect.Method.invoke(Native Method)
03-12 09:12:15.793    8648-8648/example.app W/System.err﹕ at java.lang.reflect.Method.invoke(Method.java:372)
03-12 09:12:15.793    8648-8648/example.app W/System.err﹕ at java.io.ObjectOutputStream.writeHierarchy(ObjectOutputStream.java:1033)
03-12 09:12:15.793    8648-8648/example.app W/System.err﹕ at java.io.ObjectOutputStream.writeNewObject(ObjectOutputStream.java:1384)
03-12 09:12:15.793    8648-8648/example.app W/System.err﹕ at java.io.ObjectOutputStream.writeObjectInternal(ObjectOutputStream.java:1651)
03-12 09:12:15.794    8648-8648/example.app W/System.err﹕ at java.io.ObjectOutputStream.writeObject(ObjectOutputStream.java:1497)
03-12 09:12:15.794    8648-8648/example.app W/System.err﹕ at java.io.ObjectOutputStream.writeObject(ObjectOutputStream.java:1461)
03-12 09:12:15.794    8648-8648/example.app W/System.err﹕ at java.io.ObjectOutputStream.writeFieldValues(ObjectOutputStream.java:959)
03-12 09:12:15.794    8648-8648/example.app W/System.err﹕ at java.io.ObjectOutputStream.defaultWriteObject(ObjectOutputStream.java:360)
03-12 09:12:15.794    8648-8648/example.app W/System.err﹕ at com.caverock.androidsvg.SVG$SvgConditionalContainer.writeObject(SVG.java:1864)
03-12 09:12:15.794    8648-8648/example.app W/System.err﹕ at java.lang.reflect.Method.invoke(Native Method)
03-12 09:12:15.794    8648-8648/example.app W/System.err﹕ at java.lang.reflect.Method.invoke(Method.java:372)
03-12 09:12:15.794    8648-8648/example.app W/System.err﹕ at java.io.ObjectOutputStream.writeHierarchy(ObjectOutputStream.java:1033)
03-12 09:12:15.794    8648-8648/example.app W/System.err﹕ at java.io.ObjectOutputStream.writeNewObject(ObjectOutputStream.java:1384)
03-12 09:12:15.794    8648-8648/example.app W/System.err﹕ at java.io.ObjectOutputStream.writeObjectInternal(ObjectOutputStream.java:1651)
03-12 09:12:15.794    8648-8648/example.app W/System.err﹕ at java.io.ObjectOutputStream.writeObject(ObjectOutputStream.java:1497)
03-12 09:12:15.794    8648-8648/example.app W/System.err﹕ at java.io.ObjectOutputStream.writeObject(ObjectOutputStream.java:1461)
03-12 09:12:15.794    8648-8648/example.app W/System.err﹕ at com.caverock.androidsvg.SVG.convertToPushPopOutputStream(SVG.java:176)
03-12 09:12:15.794    8648-8648/example.app W/System.err﹕ at com.caverock.androidsvg.SVG.getFromResource(SVG.java:282)
03-12 09:12:15.794    8648-8648/example.app W/System.err﹕ at com.caverock.androidsvg.SVG.getFromResource(SVG.java:264)
03-12 09:12:15.794    8648-8648/example.app W/System.err﹕ at example.app.KnooppunterFragment$KnooppunterLocationsAdapter.getView(KnooppunterFragment.java:327)
03-12 09:12:15.794    8648-8648/example.app W/System.err﹕ at android.widget.AbsListView.obtainView(AbsListView.java:2344)
03-12 09:12:15.794    8648-8648/example.app W/System.err﹕ at android.widget.ListView.makeAndAddView(ListView.java:1864)
03-12 09:12:15.794    8648-8648/example.app W/System.err﹕ at android.widget.ListView.fillDown(ListView.java:698)
03-12 09:12:15.794    8648-8648/example.app W/System.err﹕ at android.widget.ListView.fillFromTop(ListView.java:759)
03-12 09:12:15.794    8648-8648/example.app W/System.err﹕ at android.widget.ListView.layoutChildren(ListView.java:1659)
03-12 09:12:15.794    8648-8648/example.app W/System.err﹕ at android.widget.AbsListView.onLayout(AbsListView.java:2148)
03-12 09:12:15.794    8648-8648/example.app W/System.err﹕ at android.view.View.layout(View.java:15596)
03-12 09:12:15.794    8648-8648/example.app W/System.err﹕ at android.view.ViewGroup.layout(ViewGroup.java:4966)
03-12 09:12:15.794    8648-8648/example.app W/System.err﹕ at android.widget.FrameLayout.layoutChildren(FrameLayout.java:573)
03-12 09:12:15.794    8648-8648/example.app W/System.err﹕ at android.widget.FrameLayout.onLayout(FrameLayout.java:508)
03-12 09:12:15.794    8648-8648/example.app W/System.err﹕ at android.view.View.layout(View.java:15596)
03-12 09:12:15.794    8648-8648/example.app W/System.err﹕ at android.view.ViewGroup.layout(ViewGroup.java:4966)
03-12 09:12:15.794    8648-8648/example.app W/System.err﹕ at android.widget.LinearLayout.setChildFrame(LinearLayout.java:1703)
03-12 09:12:15.794    8648-8648/example.app W/System.err﹕ at android.widget.LinearLayout.layoutVertical(LinearLayout.java:1557)
03-12 09:12:15.794    8648-8648/example.app W/System.err﹕ at android.widget.LinearLayout.onLayout(LinearLayout.java:1466)
03-12 09:12:15.794    8648-8648/example.app W/System.err﹕ at android.view.View.layout(View.java:15596)
03-12 09:12:15.794    8648-8648/example.app W/System.err﹕ at android.view.ViewGroup.layout(ViewGroup.java:4966)
03-12 09:12:15.794    8648-8648/example.app W/System.err﹕ at android.widget.FrameLayout.layoutChildren(FrameLayout.java:573)
03-12 09:12:15.794    8648-8648/example.app W/System.err﹕ at android.widget.FrameLayout.onLayout(FrameLayout.java:508)
03-12 09:12:15.794    8648-8648/example.app W/System.err﹕ at android.widget.ScrollView.onLayout(ScrollView.java:1502)
03-12 09:12:15.794    8648-8648/example.app W/System.err﹕ at android.view.View.layout(View.java:15596)
03-12 09:12:15.795    8648-8648/example.app W/System.err﹕ at android.view.ViewGroup.layout(ViewGroup.java:4966)
03-12 09:12:15.795    8648-8648/example.app W/System.err﹕ at android.widget.FrameLayout.layoutChildren(FrameLayout.java:573)
03-12 09:12:15.795    8648-8648/example.app W/System.err﹕ at android.widget.FrameLayout.onLayout(FrameLayout.java:508)
03-12 09:12:15.795    8648-8648/example.app W/System.err﹕ at android.view.View.layout(View.java:15596)
03-12 09:12:15.795    8648-8648/example.app W/System.err﹕ at android.view.ViewGroup.layout(ViewGroup.java:4966)
03-12 09:12:15.795    8648-8648/example.app W/System.err﹕ at android.support.v4.view.ViewPager.onLayout(ViewPager.java:1594)
03-12 09:12:15.795    8648-8648/example.app W/System.err﹕ at android.view.View.layout(View.java:15596)
03-12 09:12:15.795    8648-8648/example.app W/System.err﹕ at android.view.ViewGroup.layout(ViewGroup.java:4966)
03-12 09:12:15.795    8648-8648/example.app W/System.err﹕ at android.widget.LinearLayout.setChildFrame(LinearLayout.java:1703)
03-12 09:12:15.795    8648-8648/example.app W/System.err﹕ at android.widget.LinearLayout.layoutVertical(LinearLayout.java:1557)
03-12 09:12:15.795    8648-8648/example.app W/System.err﹕ at android.widget.LinearLayout.onLayout(LinearLayout.java:1466)
03-12 09:12:15.795    8648-8648/example.app W/System.err﹕ at android.view.View.layout(View.java:15596)
03-12 09:12:15.795    8648-8648/example.app W/System.err﹕ at android.view.ViewGroup.layout(ViewGroup.java:4966)
03-12 09:12:15.795    8648-8648/example.app W/System.err﹕ at android.widget.FrameLayout.layoutChildren(FrameLayout.java:573)
03-12 09:12:15.795    8648-8648/example.app W/System.err﹕ at android.widget.FrameLayout.onLayout(FrameLayout.java:508)
03-12 09:12:15.795    8648-8648/example.app W/System.err﹕ at android.view.View.layout(View.java:15596)
03-12 09:12:15.795    8648-8648/example.app W/System.err﹕ at android.view.ViewGroup.layout(ViewGroup.java:4966)
03-12 09:12:15.795    8648-8648/example.app W/System.err﹕ at com.android.internal.widget.ActionBarOverlayLayout.onLayout(ActionBarOverlayLayout.java:494)
03-12 09:12:15.795    8648-8648/example.app W/System.err﹕ at android.view.View.layout(View.java:15596)
03-12 09:12:15.795    8648-8648/example.app W/System.err﹕ at android.view.ViewGroup.layout(ViewGroup.java:4966)
03-12 09:12:15.795    8648-8648/example.app W/System.err﹕ at android.widget.FrameLayout.layoutChildren(FrameLayout.java:573)
03-12 09:12:15.795    8648-8648/example.app W/System.err﹕ at android.widget.FrameLayout.onLayout(FrameLayout.java:508)
03-12 09:12:15.795    8648-8648/example.app W/System.err﹕ at android.view.View.layout(View.java:15596)
03-12 09:12:15.795    8648-8648/example.app W/System.err﹕ at android.view.ViewGroup.layout(ViewGroup.java:4966)
03-12 09:12:15.795    8648-8648/example.app W/System.err﹕ at android.view.ViewRootImpl.performLayout(ViewRootImpl.java:2072)
03-12 09:12:15.795    8648-8648/example.app W/System.err﹕ at android.view.ViewRootImpl.performTraversals(ViewRootImpl.java:1829)
03-12 09:12:15.795    8648-8648/example.app W/System.err﹕ at android.view.ViewRootImpl.doTraversal(ViewRootImpl.java:1054)
03-12 09:12:15.795    8648-8648/example.app W/System.err﹕ at android.view.ViewRootImpl$TraversalRunnable.run(ViewRootImpl.java:5779)
03-12 09:12:15.795    8648-8648/example.app W/System.err﹕ at android.view.Choreographer$CallbackRecord.run(Choreographer.java:767)
03-12 09:12:15.795    8648-8648/example.app W/System.err﹕ at android.view.Choreographer.doCallbacks(Choreographer.java:580)
03-12 09:12:15.795    8648-8648/example.app W/System.err﹕ at android.view.Choreographer.doFrame(Choreographer.java:550)
03-12 09:12:15.795    8648-8648/example.app W/System.err﹕ at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:753)
03-12 09:12:15.795    8648-8648/example.app W/System.err﹕ at android.os.Handler.handleCallback(Handler.java:739)
03-12 09:12:15.795    8648-8648/example.app W/System.err﹕ at android.os.Handler.dispatchMessage(Handler.java:95)
03-12 09:12:15.795    8648-8648/example.app W/System.err﹕ at android.os.Looper.loop(Looper.java:135)
03-12 09:12:15.795    8648-8648/example.app W/System.err﹕ at android.app.ActivityThread.main(ActivityThread.java:5221)
03-12 09:12:15.795    8648-8648/example.app W/System.err﹕ at java.lang.reflect.Method.invoke(Native Method)
03-12 09:12:15.796    8648-8648/example.app W/System.err﹕ at java.lang.reflect.Method.invoke(Method.java:372)
03-12 09:12:15.796    8648-8648/example.app W/System.err﹕ at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:899)
03-12 09:12:15.796    8648-8648/example.app W/System.err﹕ at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:694)
```

**Comment by wonson** [comment]
It seems because the child android.graphics.Matrix of GraphicsElement is not implementing Serializable protocol. So we must use "transient" prefix for all Matrix variables.

But if I do that, the variable will not be cached to stream automatically, we need to do it in a tricky way then.

Thanks for your report anyway.

**Comment by Svantulden** [comment]
Np, and thanks for your fast response. Would [anything like this](http://stackoverflow.com/a/26439672/1437096) maybe help? To replace `Matrix` with custom overridden `SerializableMatrix` which is serializable. Just an idea, as I don't know much about the inner workings.

**Comment by wonson** [comment]
Yes, of coz. Becoz it's the same solution with my link lol

**Comment by Svantulden** [comment]
Yeah sorry, I didn't see your edit with the link until after I posted my comment :P. I submitted a PR with the fix here: #2
